activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,13 @@
+*   Allow configuring if Postgres structure dumps include comments
```

Comment:
My thinking is that disabling comments by default in 7.0 was a mistake. One way to minimise the amount of time that the behaviour is "wrong" could be to backport this to `7-0-stable`.

In terms of how disruptive this is:

- If you rely on comments on columns (anecdotally I haven't seen this much in the wild and we've used postgres in production for 10 years, *but* at least 1 user did report it), there's now no way to get them in your structure dump. I don't think this will break anyone's app, but those comments are presumably useful so we shouldn't remove them on your behalf.
- The issue described in https://github.com/rails/rails/issues/43107 would only happen if you're using an extension and you're running CI in an environment that lacks superuser (or I guess if you're loading a development environment from a structure dump in a restricted environment?). Using extensions is pretty common, but I'd hope that whatever CI you're running in you can add superuser if you need it. Again, and at least 1 user has reported it.

In short, I don't think it's a huge disruption because there's workarounds either way. But we should try and make the issue go away.

Reply:
> we just revert https://github.com/rails/rails/pull/43216 and push people to https://github.com/rails/rails/issues/43107#issuecomment-906573620

Yeah, that's my inclination. There are already multiple valid workarounds here, so no need for us to invent an additional almost-but-not-quite similar means of setting that option. (I believe the upstream-recommended workaround would be to run psql in default mode, where it will not abort upon the ignorable comment-setting error, and instead continue happily.)

This is really a Postgres bug, though I don't know if it's been discussed upstream there. Comments on extensions get created as a side-effect of adding an extension, which by any reasonable definition means they belong to the extension, and therefore [just like the extension's schema, tables, functions, etc] should not be separately [and redundantly] dumped by pg_dump.

Reply:
Okay, I can make a PR for that.

Reply:
https://github.com/rails/rails/pull/44633